### PR TITLE
Small interface improvement in ActionWarehouse::getActionListByName

### LINKED ISSUE #13004: Small interface function change in ActionWarehouse
## Rationale

Calling `ActionWarehouse::getActionListByName` could fail if there is no actions on the task passed in. I think it is better to just return an empty list.

## Description

The reason I am asking this is that I have bunch of places calling this function in my application and I have to call `ActionWarehouse::hasActions` before calling this function, which is quite tedious. Worst part is that sometimes I forgot calling `hasActions`, which works fine at a moment but could potentially error out with some unrelated changes like removing an action on the task. I think returning an empty list is fine. If you think this is ok, I can submit a small PR quickly.

## Impact
A small change that make the code less tedious.

### PR DESCRIPTION
**Pull request by YaqiWang**
Close #13004.

### CODE REVIEW

**Review by permcody — DISMISSED** [review]
Hmm... I'm not against this, but documentation probably needs to be updated. Also, what is the use case?

**Comment by YaqiWang** [comment]
Added one extra comments in the header file. This function is used in Rattlesnake to obtain non-const pointers of Actions. The rational for this change is provided in the issue.

File: framework/include/actions/ActionWarehouse.h [hunk 1]
```diff
@@ -99,6 +99,7 @@ class ActionWarehouse : public ConsoleStreamInterface
 
   /**
    * Retrieve a constant list of \p Action pointers associated with the passed in task.
+   * Empty list will be returned if no actions are asslcoated with the task.
```
Comment (permcody) [inline framework/include/actions/ActionWarehouse.h]: asslcoated -> associated

Comment (YaqiWang) [reply framework/include/actions/ActionWarehouse.h]: how this can be???

**Review by permcody — DISMISSED** [review]

**Comment by YaqiWang** [comment]
I checked moose and found one use case. This will ensure an inside-moose check.

File: framework/src/actions/CheckOutputAction.C [hunk 2]
```diff
@@ -41,27 +41,24 @@ CheckOutputAction::act()
 void
 CheckOutputAction::checkVariableOutput(const std::string & task)
 {
-  if (_awh.hasActions(task))
+  // Loop through the actions for the given task
```
Comment (YaqiWang) [inline framework/src/actions/CheckOutputAction.C]: I hope the differ can be smarter to show that only indentation is different here and do not break the block into pieces...

Comment (permcody) [reply framework/src/actions/CheckOutputAction.C]: Nope!

Comment (permcody) [reply framework/src/actions/CheckOutputAction.C]: Actually it does, but not by default. You can hide whitespace changes in the "Diff settings".

**Review by permcody — APPROVED** [review]
Hmm... I'm not against this, but documentation probably needs to be updated. Also, what is the use case?

**Comment by YaqiWang** [comment]
Ping @permcody

**Pull request merged**